File: companion/spss/python/b09.py
import pandas as pd
import numpy as np
from scipy import stats

veri = pd.read_csv("companion/spss/csv/b09.csv")
assert veri.shape == (395, 10)
assert not veri.isna().any().any()
assert veri.pass10.eq((veri.g3 >= 10).astype(int)).all()
gozlem_sayisi = len(veri)
ortalama = veri.g3.mean()
standart_sapma = veri.g3.std(ddof=1)
standart_hata = standart_sapma / np.sqrt(gozlem_sayisi)
sonuc = stats.ttest_1samp(
    veri.g3, popmean=10, alternative="two-sided")
ortalama_alt, ortalama_ust = stats.t.interval(
    .95, df=gozlem_sayisi-1,
    loc=ortalama, scale=standart_hata)
ozet = pd.Series({
    "n": gozlem_sayisi, "ort": ortalama, "ss": standart_sapma,
    "sh_ort": standart_hata, "t": sonuc.statistic,
    "df": gozlem_sayisi-1, "p_iki_yonlu": sonuc.pvalue,
    "fark": ortalama-10,
    "ort_alt": ortalama_alt, "ort_ust": ortalama_ust,
    "fark_alt": ortalama_alt-10, "fark_ust": ortalama_ust-10
})
print(ozet.to_string(float_format=lambda deger: f"{deger:.13f}"))
print(int(veri.g3.eq(0).sum()))

File: companion/spss/r/b09.R
veri <- read.csv("companion/spss/csv/b09.csv")
stopifnot(nrow(veri) == 395, ncol(veri) == 10,
          !anyNA(veri),
          all(veri$pass10 == as.integer(veri$g3 >= 10)))
sonuc <- t.test(veri$g3, mu=10,
               alternative="two.sided", conf.level=.95)
ozet <- c(
  n=nrow(veri), ort=mean(veri$g3), ss=sd(veri$g3),
  sh_ort=sd(veri$g3)/sqrt(nrow(veri)),
  t=unname(sonuc$statistic), df=unname(sonuc$parameter),
  p_iki_yonlu=sonuc$p.value, fark=mean(veri$g3)-10,
  ort_alt=sonuc$conf.int[1], ort_ust=sonuc$conf.int[2],
  fark_alt=sonuc$conf.int[1]-10,
  fark_ust=sonuc$conf.int[2]-10)
print(sonuc)
print(ozet, digits=15)
print(sum(veri$g3 == 0))